pkg/hubble/parser/seven/parser.go

```diff
@@ -129,6 +129,7 @@ func (p *Parser) Decode(r *accesslog.LogRecord, decoded *pb.Flow) error {
 	decoded.Time = pbTimestamp
 	decoded.Verdict = decodeVerdict(r.Verdict)
 	decoded.DropReason = 0
+	decoded.DropReasonDesc = pb.DropReason_SUCCESS
```

Comment:
`drop reason success` is rather confusing. Zero value should be left as some sort of an `UNKNOWN` if it's not applicable to the current type.

Besides, you don't have to set it explicitly.

Reply:
i guess this is trying to be consistent with https://github.com/cilium/cilium/blob/b3adc4d9410c76db934df021f99acaa033adf50f/pkg/monitor/api/drop.go#L27-L86

kind of curious how it shows up with `hubble observe -o json` though. will it show up like this?
```
"drop_reason_desc": "SUCCESS"
```
that would be a bit strange.

Reply:
Maybe we should omit non-drop reasons from the enum, that way `drop_reason_desc` will be a missing field in theses cases. Would that make sense?

Reply:
We have to be careful with the zero (default) value. I'd prefer if we used `DROP_REASON_UNKNOWN = 0` to future proof it. According to this [post](http://androiddevblog.com/protocol-buffers-pitfall-adding-enum-values/), we ever add more drop reasons, client with older protobuf schemas will use the zero-valued enum to represent it. 

In that case, we want to avoid displaying `drop_reason: SUCCESS`. However, as you pointed out correctly, the zero value is treated as an missing value anyways, and not be displayed in the JSON output by default.

Reply:
Thinking about this some more, while it won't show up in JSON output by default, I think we should either name it `DROP_REASON_NONE` or `DROP_REASON_UNKNOWN`.

Calling it `SUCCESS` is a very leaky abstraction (it's originating in the kernel's errno `0`) and confusing to users of the API (they should not have to know or care about this).

Reply:
+1 for `DROP_REASON_NONE`. `DROP_REASON_UNKNOWN` sounds a bit like the flow was dropped but the reason is unknown :)

Reply:
> [...] In these cases, we want the field to describe it as DROP_REASON_UNKNOWN, and not SUCCESS (which is wrong, as the flow was dropped).

@gandro my understanding is that values [<130 are "non-drop reasons"](https://github.com/cilium/cilium/blob/master/pkg/monitor/api/drop.go#L21,L22) and so I don't think they can show up when the flow was dropped.

Reply:
Okay, there are two independent issues here:

### Issue 1: The meaning of `non-drop` drop-reasons

There are two kinds of BPF perf events relevant to Hubble which have the numeric drop reason populated: `DropNotify` events, and `PolicyVerdict` events with a negative verdict. 

The `DropMin` is used for the [`reason` field the metricsmap](https://github.com/cilium/cilium/blob/ada53a91a98618d24dad2abda97dd9c60923003c/bpf/lib/common.h#L248), which is overloaded to contain both drop reasons and non-drop-reasons. The metricsmap (not to be confused with the Hubble metrics) are irrelevant to Hubble at this point. 

The values that Hubble _can_ observe are the ones that the datapath produces for `DropNotify` and `PolicyVerdict` events, which are defined here:

https://github.com/cilium/cilium/blob/ada53a91a98618d24dad2abda97dd9c60923003c/bpf/lib/common.h#L327-L382

Therefore, I say we remove all the variants below 130 from the enum. The `errors` map in Go mixes up drop reasons and non-drop reasons simply for the [`DropForwardReason`](https://github.com/cilium/cilium/blob/fb0c3e592d7a9da854f9d38345fbb5720ea3b073/pkg/maps/metricsmap/metricsmap.go#L141) in the metricsmap. Again, nothing that is relevant to Hubble.

### Issue 2: The special meaning of a zero-valued enum in protobuf. 

 proto3 does not distinguish between an absent value and the zero-valued variant. This means that whenever you will use this new protobuf definition to talk to an older implementation of Hubble, *all* flows (dropped flows as well as forwarded l7 flows) will have `drop_reason_desc: SUCCESS` (even if it is not displayed in the JSON output because it's a zero-valued variant, it is still the value used in the Go API)

I don't think `drop_reason_desc: SUCCESS` makes sense for either forwarded or dropped flows. Let's quickly look at some the two cases:

1. Decoding a drop event from old Hubble implementation (which does not populate `drop_reason_desc`): `drop_reason` will be non-zero, but `drop_reason_desc` will be  `SUCCESS`. That's simply plain wrong. Here, `DROP_REASON_NONE` is also kinda wrong here (we do have a drop reason, but we don't know it's description), which is why I favor `DROP_REASON_UNKNOWN` over `DROP_REASON_NONE`.

2. Decoding a forwarded event (e.g. L7 or TraceNotify): `drop_reason_desc: SUCCESS` arguably makes some sense, but it's still confusing. What does it mean for a `drop` to be successful in a flow which was not dropped? Arguably, `DROP_REASON_NONE` would be the most correct here, `DROP_REASON_UNKNOWN` is at least not wrong.

### TL;DR: I propose the following changes:

1. Remove all non-drop reason variants from the enum (values below 130).
2. Let's name the zero-valued variant something else than `SUCCESS`. Either `DROP_REASON_UNKNOWN` (preferred by me) or `DROP_REASON_NONE`.